buildSrc/src/main/java/io/opentelemetry/instrumentation/gradle/bytebuddy/ByteBuddyPluginConfigurator.java

```diff
@@ -68,6 +68,7 @@ public void configure() {
   private Task createLanguageTask(AbstractCompile compileTask, String name) {
     ByteBuddySimpleTask task = project.getTasks().create(name, ByteBuddySimpleTask.class);
     task.setGroup("Byte Buddy");
+    task.getOutputs().cacheIf(unused -> true);
```

Comment:
It is indeed suspicious. I would assume that better approach would be to have correctly configured inputs and outputs for this task

Reply:
Do you mean to enable caching? The task is not a `CacheableTask` so even with correct input / outputs configured it can't be cached without opting in with this command.

I'm concerned about the determinism or the task but the outputs inputs seem sane from what I can tell.

Reply:
To clarify, UP-TO-DATE seems to be working fine. What doesn't work

- After a clean, this task always reruns since it's not a `CacheableTask`. We can fix that by adding this line.

- If the task reruns after a clean, all downstream tasks also rerun even though we would expect deterministic output meaning this wouldn't happen. That needs more investigation in #1765.

I think this change is still worth it even if we fix the determinism issue, and maybe we can survive some risk of cache pollution to get the build sane again. Let me try rerunning the tests in this PR and see

Reply:
Reran tests, cache worked fine-ish. Build / Java 8 3min, Java 15 25min - still something obviously strange about Java 15 but this seems to at least help us well.